service/history/task/cross_cluster_source_task_executor.go

```diff
@@ -269,6 +265,113 @@ func (t *crossClusterSourceTaskExecutor) executeCancelExecutionTask(
 	}
 }
 
+func (t *crossClusterSourceTaskExecutor) getCommonError(
+	ctx context.Context,
+	wfContext execution.Context,
+	mutableState execution.MutableState,
+	task *crossClusterSourceTask,
+) error {
+	// handle common errors
+	failedCause := task.response.FailedCause
+	if failedCause != nil {
+		switch *failedCause {
+		case types.CrossClusterTaskFailedCauseDomainNotActive:
+			switch err := t.generateNewTask(ctx, wfContext, mutableState, task); {
+			case err != nil:
+				return err
+			default:
+				return errNewTask
+			}
+		case types.CrossClusterTaskFailedCauseWorkflowNotExists:
+			return &types.EntityNotExistsError{}
+		case types.CrossClusterTaskFailedCauseWorkflowAlreadyCompleted:
+			return &types.WorkflowExecutionAlreadyCompletedError{}
+		case types.CrossClusterTaskFailedCauseUncategorized:
+			// ideally this case should not happen
+			// crossClusterSourceTask.Update() will reject response with this failed cause
+			t.setTaskState(task, ctask.TaskStatePending, processingState(task.response.TaskState))
+			return errContinueExecution
+		}
+	}
+	return nil
+}
+
+func (t *crossClusterSourceTaskExecutor) executeApplyParentClosePolicyTask(
+	ctx context.Context,
+	task *crossClusterSourceTask,
+) (retError error) {
+	taskInfo := task.GetInfo().(*persistence.CrossClusterTaskInfo)
+	wfContext, mutableState, release, err := loadWorkflowForCrossClusterTask(ctx, t.executionCache, taskInfo, t.metricsClient, t.logger)
+	if err != nil || mutableState == nil {
+		return err
+	}
+	defer func() { release(retError) }()
+
+	err = task.VerifyTaskFailoverVersion(mutableState, taskInfo)
```

Comment:
same comment as the one in `cross_cluster_task.go`, if `verifyTaskVersion` returned inside this util method returned `false, nil`, we should also return nil here to skip the task.